### DISCUSSION TITLE
Include "Battery Install" button...

### DISCUSSION THREAD

**Discussion by smoketabs** [nathanmarlor/foxess_em#313] [Ideas]
Hi.

For those of us who, from time to time, will be increasing the battery stacks as and when funds allow, I was wondering if it's possible to have toggle/button available that, once toggled, sets the appropriate registers (and algorithm) to get the existing install to 50% SoC? Once the battery stack install is complete, the user would then un-toggle and then be prompted to adjust the relavent EM config. fields to reflect the new capacity and battery charge/voltage details only?

Just a thought.

Thanks.

**Comment by canton7** [comment]
The thing is, I'm not sure that there's a set of registers you can set to get the battery to 50%.

* If the battery is above 50%, the only way to drain it is by using electricity from it.
* If the battery is below 50%, we can charge it up to 50% from the grid, but nothing stops it charging further from solar.

So, I think any attempt needs input from the user, to turn off the DC isolator at the appropriate point.

Unless I'm missing a trick, of course.

**Reply by smoketabs** [reply]
I wonder if there's a register to set MAX SoC? Then there're the registers to set MIN SoC and MIN SoC (On Grid). Or I might be missing something (more than likely!)

However, Is there a way to "pause" the EM? I ask as if I manually set the SoCs, I don't want the EM to set them back to what ever they deem best until after I've installed a new battery. Once I have installed a new battery, I can then resume EM. I know there's a "toggle" but does that actually pause EM?
![Screenshot 2023-06-24 at 09 07 34](https://github.com/nathanmarlor/foxess_em/assets/15179147/34901063-c3b3-4c9e-b6b7-4085d63f7edd)

**Reply by canton7** [reply]
There is a max soc, however it only applies to the grid and does not stop the batteries charging from solar. Even if you set the work mode to Feed-in First, if you exceed your export limit your batteries will charge fom solar. This is why I say I don't think there's a way to reliably keep the registers at 50% without the user manually doing things to the inverter.
